src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -1300,24 +1300,50 @@ public void addChannel(ChannelShim channelShim)
      */
     public void removeChannel(ChannelShim channelShim)
     {
-        synchronized (channelShims)
+        if (channelShims.remove(channelShim))
         {
-            switch (channelShim.getMediaType())
+            if (channelShims.isEmpty())
```

Comment:
the `remove`/`isEmpty` check/accept video/audio before would've been atomic and now it's not.  are we sure that's not still needed?

Reply:
I could bring back `synchronized` section, if it purpose was to introduce atomic behavior between `remove` and `isEmpty`. But I'm actually can't figure out which problem it solved before? Currently ` if (channelShims.remove(channelShim))` is necessary to check if `channelShims` was modified during `remove` - it is optional check which could be omimted. The `if (channelShims.isEmpty())` necessary to check if there is no currently channels associated with endpoint.
As far as I understand endpoint become expired as soon as all channels removed.

Reply:
the concern would be in the remove scenario if we ever failed to perform expiration when we should have.  i _think_ we're safe here without the sync block, though.  we are slightly more susceptible to the case where the remove gets run before an add (for the same channel) though, i think--the proper answer there would be to make sure we handle those calls from a single thread, as otherwise it's always possible to some degree.